test/integration/LockRepoSpec.scala

```diff
@@ -0,0 +1,180 @@
+
+  "LockRepo" should {
+
+    "when requesting a lock for a client subscription Id should return true if lock does not already exist" in {
+      val csId = UUID.randomUUID()
+      val ownerId = LockOwnerId("caller1")
+
+      await(lockRepo.tryToAcquireOrRenewLock(ClientSubscriptionId(csId), ownerId, fiveSecondsDuration)) shouldBe true
+    }
+
+    "when requesting a lock for a client subscription Id should return true even if lock already exists" in {
+      val csId = UUID.randomUUID()
+      val ownerId = LockOwnerId("caller1")
+
+      await(lockRepo.tryToAcquireOrRenewLock(ClientSubscriptionId(csId), ownerId, twentyFiveSecondsDuration)) shouldBe true
+      await(lockRepo.tryToAcquireOrRenewLock(ClientSubscriptionId(csId), ownerId, twentyFiveSecondsDuration)) shouldBe true
+    }
+
+    "when requesting a lock should return false if we do not own the lock" in {
+      val csId = UUID.randomUUID()
+      val ownerId1 = LockOwnerId("caller1")
+      val ownerId2 = LockOwnerId("caller2")
+
+      await(lockRepo.tryToAcquireOrRenewLock(ClientSubscriptionId(csId), ownerId1, twentyFiveSecondsDuration)) shouldBe true
+      await(lockRepo.tryToAcquireOrRenewLock(ClientSubscriptionId(csId), ownerId1, twentyFiveSecondsDuration)) shouldBe true
+      await(lockRepo.tryToAcquireOrRenewLock(ClientSubscriptionId(csId), ownerId2, twentyFiveSecondsDuration)) shouldBe false
+    }
+
+    "when requesting to release a lock that exists and is owned by caller, should release successfully" in {
+      val csId = ClientSubscriptionId(UUID.randomUUID())
+      val ownerId1 = LockOwnerId("worker1")
+      val ownerId2 = LockOwnerId("worker2")
+
+      await(lockRepo.tryToAcquireOrRenewLock(csId, ownerId1, twentyFiveSecondsDuration)) shouldBe true
+      await(lockRepo.release(csId, ownerId1)) should be ((): Unit)
+      await(lockRepo.tryToAcquireOrRenewLock(csId, ownerId2, twentyFiveSecondsDuration)) shouldBe true
+    }
+
+    "when requesting to release a lock that exists and is NOT owned by caller, should fail to release lock" in {
+      val csId = ClientSubscriptionId(UUID.randomUUID())
+      val ownerId1 = LockOwnerId("worker1")
+      val ownerId2 = LockOwnerId("worker2")
+
+      await(lockRepo.tryToAcquireOrRenewLock(csId, ownerId1, twentyFiveSecondsDuration)) shouldBe true
+      await(lockRepo.release(csId, ownerId2)) should be ((): Unit)
+      await(lockRepo.tryToAcquireOrRenewLock(csId, ownerId2, twentyFiveSecondsDuration)) shouldBe false
+    }
+
+    "when requesting to refresh a lock that exists and is owned by caller, should refresh successfully" in {
+      val csId = ClientSubscriptionId(UUID.randomUUID())
+      val ownerId1 = LockOwnerId("worker1")
+
+      await(lockRepo.tryToAcquireOrRenewLock(csId, ownerId1, twentyFiveSecondsDuration)) shouldBe true
+      await(lockRepo.tryToAcquireOrRenewLock(csId, ownerId1, twentyFiveSecondsDuration)) shouldBe true
```

Comment:
Need to make sure that expiry time is correct in case of refresh

Reply:
Not sure what you mean? in this instance we are just refreshing the the lock. are you wanting to pull out the locks and check the expiry time from the db? This is currently not returned so I would need to make code changes to return this information just to test this.